source/core.py

```diff
@@ -163,7 +170,12 @@ def resetConfiguration(factoryDefaults=False):
 	config.conf.reset(factoryDefaults=factoryDefaults)
 	logHandler.setLogLevelFromConfig()
 	#Language
-	lang = config.conf["general"]["language"]
+	lang = globalVars.appArgs.cmdLineLanguage
+	if lang:
+		# Ensure the language specified on the command line will be saved with the config.

```

Comment:
I"m not sure whether this is what people expect from command line arguments. They are most of the time only seen as temporary overrides. I think I prefer the language setting to be that way as well, i.e. it should only be effective to the current session of NVDA, until there is a restart triggered from outside that resets the params. It should not touch the configuration at all.

Reply:
> What happens if you don't provide the parameter and change the language when save on exit is disabled? Is it then properly updated?

(Before and after this PR, same process)
If one changes the language and answers "No" when asked to restart right away, the change is - as expected - lost when restarting if the configuration has not been saved.

Reply:
> It should not touch the configuration at all.

The original poster of #10144 accidentally changed language to Hebrew, a language which he cannot understand. To get him back on his feet, we need a way to change this setting.

The point is: If the language specified on the command line is different from the configuration, what should the settings dialog present as the selected language?

I see four options here:
1. The language specified on the command line. Hence, if the configuration is saved (manually or automatically upon exit), the new language is applied. This is the approach I took.
2. The original language of the configuration, which does not reflect the state NVDA is currently in. I felt it was quite awkward when testing.
3. The language specified on the command line suffixed with "forced from command line". This is explicit, but also requires greater impact and I felt it was uselessly cumbersome.
4. The original language of the configuration, but in a disabled drop-down preceded by a message explaining another language was specified on the command line and a button to allow for changing this setting. This is explicit, but also requires the greatest impact and I felt it was uselessly cumbersome.

What do you think? Would you prefer option 2?

Reply:
In other situations where we override from the command-line we disable the control so that we avoid this confusion. That, of course, defeats the purpose of this PR.

My preference would be to insert a new item in the language list (at index 0) with something like `command line option: en_au` or similar. 
- If the user leaves this option, they can safely change other options and save their config without accidentally changing their language.
- If the user explicitly changes the language to `en_au` then we know they wish to make it permanent.